packages/build/config/tsconfig.common.json

```diff
@@ -10,6 +10,8 @@
     // FIXME(bajtos) LB4 is not compatible with this setting yet
     "strictPropertyInitialization": false,
 
+    "incremental": true,
+
```

Comment:
Please add `"tsBuildInfoFile": ".tsbuildinfo",`.

Reply:
AFAIK, "tsBuildInfoFile": ".tsbuildinfo"` setting in `packages/build/config/tsconfig.common.json` will be resolved as `packages/build/config/.tsbuildinfo`, which isn't what we want.

I think we will have to add `tsBuildInfoFile` setting to every per-package `package.json` file.

I'll take a look.

Reply:
IIRC, it's resolved against the `outDir`.

Reply:
Quoting from https://www.typescriptlang.org/docs/handbook/tsconfig-json.html#configuration-inheritance-with-extends:

> All relative paths found in the configuration file will be resolved relative to the configuration file they originated in.

See also https://github.com/Microsoft/TypeScript/issues/29172

Quoting also from https://github.com/Microsoft/TypeScript/pull/29813

> * When tsconfig specifies compiler options `incremental` file with `.tsbuildinfo` extension is emitted.
> * Composite projects indirectly specify `incremental` as `true` and its error to specify `incremental` as `false` in the composite project
> * If project specifies `tsBuildInfoFile` options, it uses the path specified by that to write the build information.
> * If project specifies `out` or `outFile` option, the `outFileWithoutExtension.tsbuildinfo` file is written next to output js file.
> * If project specified `outDir`, config files base file name with extension as `.tsbuildinfo` is written in `outDir`
> * Otherswise config files base file name with extension as `.tsbuildinfo` is written next to the config file

AFAICT, we are not setting `outDir` in our `tsconfig` files and configure it via a CLI option instead. This is a legacy from the days where we used to have multiple dist directories (`dist8` for Node.js 8, `dist10` for Node.js 10, etc.)

Reply:
> If project specified `outDir`, config files base file name with extension as `.tsbuildinfo` is written in `outDir`

This rule does not seem to be triggered in our current build setup, even when I change all `tsconfig.build.json` files to include `"outDir": "dist"` :(